ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala

```diff
@@ -0,0 +1,107 @@
+package com.daml.ledger.participant.state.kvutils
+
+import com.daml.ledger.participant.state.kvutils.Conversions.{
+  absoluteContractIdToLogEntryId,
+  absoluteContractIdToStateKey,
+  relativeContractIdToStateKey
+}
+import com.daml.ledger.participant.state.v1.SubmittedTransaction
+import com.digitalasset.daml.lf.transaction.GenTransaction
+import com.digitalasset.daml.lf.transaction.Node.{
+  NodeCreate,
+  NodeExercises,
+  NodeFetch,
+  NodeLookupByKey
+}
+import com.digitalasset.daml.lf.value.Value.{AbsoluteContractId, ContractId, RelativeContractId}
+import com.daml.ledger.participant.state.kvutils.DamlKvutils._
+
+private[kvutils] object InputsAndEffects {
+
+  /** The effects of the transaction, that is what contracts
+    * were consumed and created, and what contract keys were updated.
```

Comment:
add note that contract-keys are missing for now; and link to issue in BETA milestone.

ALSO: make sure that there is a hard-failure and an error log entry in case a contract-key is used.